tests/unit/conftest.py

```diff
@@ -135,14 +136,40 @@ def restart_savepoint(session, transaction):  # pylint:disable=unused-variable
         ):  # pylint:disable=protected-access
             session.begin_nested()
 
+    set_factory_boy_sqlalchemy_session(session)
+
     try:
         yield session
     finally:
+        clear_factory_boy_sqlalchemy_session()
         session.close()
         trans.rollback()
         conn.close()
 
 
+def sqlalchemy_factory_classes():
+    # Return all the SQLAlchemy factory classes from tests.factories.
+    for factory_class in factories.__dict__.values():
+        try:
+            is_sqla_factory = issubclass(factory_class, SQLAlchemyModelFactory)
+        except TypeError:
+            is_sqla_factory = False
+
+        if is_sqla_factory:
+            yield factory_class
+
+
+def set_factory_boy_sqlalchemy_session(session):
+    for factory_class in sqlalchemy_factory_classes():
+        # pylint:disable=protected-access
+        factory_class._meta.sqlalchemy_session = session
+
+
+def clear_factory_boy_sqlalchemy_session():
+    for factory_class in sqlalchemy_factory_classes():
+        factory_class._meta.sqlalchemy_session = None  # pylint:disable=protected-access
```

Comment:
Might be nice to have a short comment for these guys saying why we're doing this.

Reply:
Added comments

Reply:
How would you normally be expected to have a session about like that?

Reply:
They seem to expect you to have a module-level session. See their example here: https://factoryboy.readthedocs.io/en/latest/orms.html#sqlalchemy